### PR TITLE
Task#14 and adding in some extra error handling tests

### PR DESCRIPTION
**Pull request by Pimmy17**
Task#14 and added some more error handling, including reinstating a generic error handling function that can help identity future errors.

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -3,6 +3,7 @@ const app = require("../app.js");
 const seed = require("../db/seeds/seed.js");
 const data = require("../db/data/test-data");
 const db = require("../db/connection.js");
+const { expect } = require("@jest/globals");
```
Comment (daviidbartlett) [inline __tests__/app.test.js:7]: don't need this

File: __tests__/app.test.js [hunk 2]
```diff
@@ -82,4 +83,42 @@ describe('Testing app', () => {
...
         });
     });
+    describe('GET /api/articles/:article_id', () => {
+        test('status: 200, responds with a single article', () => {
+          return request(app)
+            .get(`/api/articles/1`)
+            .expect(200)
+            .then(({ body: { article } }) => {
+                article.forEach((oneArticle) => {
+                  expect(oneArticle).toEqual(
```
Comment (daviidbartlett) [inline __tests__/app.test.js:93]: shouldn't be coming back in an array! It's just one article, I'd like an object please

File: __tests__/app.test.js [hunk 3]
```diff
@@ -82,4 +83,42 @@ describe('Testing app', () => {
...
+            .then(({ body: { msg } }) => {
+                expect(msg).toBe("Wrong Input!")
+            })
+        });
+        test('status: 404, when a valid type for article id is entered, however it currently does not exist', () => {
+            return request(app)
+            .get(`/api/articles/999999`)
+            .expect(404)
+            .then(({ body: { msg } }) => {
+                expect(msg).toBe("Woah! We're not that big yet!")
```
Comment (daviidbartlett) [inline __tests__/app.test.js:120]: fun message but not helpful if article one has been deleted and then I try and get it. Just tell them that that article_id doesn't exist

File: controllers/articles.controllers.js [hunk 4]
```diff
@@ -9,4 +9,17 @@ exports.getArticles = (req, res, next) => {
...
\ No newline at end of file
+};
+
+exports.getArticleById = (req, res, next) => {
+    const {article_id} = req.params;
+    fetchArticleById(article_id)
+    .then((article) => {
+        res.status(200).send({article})
+        console.log(article)
+    })
```
Comment (daviidbartlett) [inline controllers/articles.controllers.js:20]: don't forget to clear your console.logs

File: models/articles.models.js [hunk 5]
```diff
@@ -4,7 +4,18 @@ exports.fetchArticles = () => {
     return db
     .query(`SELECT article_id, author, created_at, title, topic, votes FROM articles ORDER BY created_at DESC;`)
     .then(({ rows: articles }) => {
-        console.log(articles)
+        // console.log(articles)
         return articles;
```
Comment (daviidbartlett) [inline models/articles.models.js:8]: remove

File: models/users.models.js [hunk 6]
```diff
@@ -4,7 +4,7 @@ exports.fetchUsers = () => {
     return db
     .query(`SELECT username FROM users;`)
     .then(({ rows: users }) => {
-        console.log(users)
+        // console.log(users)
         return users;
```
Comment (daviidbartlett) [inline models/users.models.js:8]: remove

**Review by daviidbartlett — COMMENTED** [review]
just a few comments

File: __tests__/app.test.js (hunk 1 above)
Comment (Pimmy17) [reply __tests__/app.test.js:7]: Yeah I saw that and removed it. Keeps coming up if I don't tab for describe, test or expect in the test area.

File: controllers/articles.controllers.js (hunk 4 above)
Comment (Pimmy17) [reply controllers/articles.controllers.js:20]: Cleared

File: __tests__/app.test.js (hunk 3 above)
Comment (Pimmy17) [reply __tests__/app.test.js:120]: Altered to ID Does Not Exist

File: __tests__/app.test.js (hunk 2 above)
Comment (Pimmy17) [reply __tests__/app.test.js:93]: Removed the forEach and adjusted the code to bring back a single object. Test working.

File: models/articles.models.js (hunk 5 above)
Comment (Pimmy17) [reply models/articles.models.js:8]: Removed

File: models/users.models.js (hunk 6 above)
Comment (Pimmy17) [reply models/users.models.js:8]: Removed

**Pull request closed without merging**